io/src/test/scala/fs2/io/tcp/SocketSpec.scala

```diff
@@ -127,5 +128,43 @@ class SocketSpec extends Fs2Spec {
           .get
       result shouldBe sizes
     }
+
+    "write - concurrent calls do not cause WritePendingException" in {
```

Comment:
Could we simulate the conditions leading to the exception by connecting to a (fs2-io tcp-based) server that pulls bytes from the socket reaaally slow?

Reply:
really slow, as in "waits for a promise that's fulfilled some time around when the second write attempt is started" :)

Reply:
At the moment we're connecting to a server that doesn't pull bytes at all, which I guess is the limiting case of pulling them really slow. I've pushed a change to get the exception without changing the write buffer size.

Reply:
I’m not quite sure I’m following. What’s the issue with the test as it stands?

Reply:
I suppose I don't like that we rely on an arbitrary amount for the size of the buffer (I think the server one can be configured, maybe the client one can too) and the amount of concurrent requests to write. But I won't stand in the way if other maintainers are okay with it :)